modMcf/src/org/aion/mcf/config/CfgLog.java

```diff
@@ -148,9 +155,43 @@ public String toXML() {
     public boolean getLogFile() {
         return this.logFile;
     }
-    
-    /** Method checks user input path of logFile */
+
+    /** Method returns user input path of logFile */
     public String getLogPath() {
-        return this.logPath;
+        if (!isValidPath()) {
+            return this.logPath = "log";
+        } else {
+            return this.logPath;
+        }
+    }
+
+    /** Method checks logPath for illegal inputs */
+    public boolean isValidPath() {
+        if (logPath.contains("-")
+                || logPath.contains("+")
+                || logPath.contains("=")
+                || logPath.contains("?")
+                || logPath.contains("!")
+                || logPath.contains("@")
+                || logPath.contains("#")
+                || logPath.contains("$")
+                || logPath.contains("%")
+                || logPath.contains("^")
+                || logPath.contains("*")
+                || logPath.contains(":")
+                || logPath.contains(";")
+                || logPath.contains(".")
+                || logPath.contains(",")
+                || logPath.contains("'")
+                || logPath.contains("|")) {
+            return this.validPath = false;
```

Comment:
Can you use regular expression instead of your implement?
https://docs.oracle.com/javase/7/docs/api/java/util/regex/Pattern.html

Reply:
Updated condition using Regex